core/src/main/java/com/datastrato/gravitino/authorization/UserGroupManager.java

```diff
@@ -48,8 +48,18 @@ class UserGroupManager {
   }
 
   User addUser(String metalake, String name) throws UserAlreadyExistsException {
+
+    AuthorizationUtils.checkMetalakeExists(metalake);
+    AuthorizationUtils.checkPermission(
```

Comment:
It's not a best practice to checkPermission everywhere in this package, why can't you do this in a filter before the request actually executed?

Reply:
The implementation here is so broken and split into two groups of API, which makes the change so hard to maintain.

Reply:
> It's not a best practice to checkPermission everywhere in this package, why can't you do this in a filter before the request actually executed?

Some operations will execute some extra actions after the operations are executed For example, list operation. We should filter the entities which we don't have any privilege.
For example, create operation. We should create some system roles for these operations.

Reply:
It's due to your design model problem, you're not implementing a consistent role base model, which makes the code so broken, in which you differentiate service user, admin user and normal user, each user has a different implementation logic to handle.

Anyway, I don't like the current implementation, it is not a good code implementation.